controllers/fio/controller.go

```diff
@@ -60,7 +66,22 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, err
 	}
 
-	job := NewJob(&cr, configMap)
+	pvcName := cr.Spec.PersistentVolumeClaimName
+	if pvcName == nil {
+		pvc, err := NewPersistentVolumeClaim(&cr)
+		if err != nil {
+			r.K8S.EventRecorder.Eventf(crReference, corev1.EventTypeNormal, k8s.CreateSucceeded,
```

Comment:
So far we haven't reported such errors via Eventf. We can keep as-is, but I don't think it adds much value to see k8s related errors as events. Additionally, I don't think that's how k8s handles it.

Reply:
The only reason I agree with you here after reconsideration is that because of the validation this error shouldn't occur in the first place. To put it another way, the validation is the user facing error handling for this, so you're right, no need for this event.